### PR TITLE
PIM-5777: PEF attribute values not refreshed when locale is changed

### PR DESCRIPTION
**Pull request by momoss**
Refresh PEF attributes labels values for multi-select and simple-select when locale is changed.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | N
| Added Behats                      | N
| Changelog updated                 | 
| Review and 2 GTM                  |
| Micro Demo to the PO (Story only) | N
| Migration script                  | N
| Tech Doc                          | N

### CODE REVIEW

File: CHANGELOG-1.5.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## Bug fixes
 
+- PIM-5777: PEF attribute values not refreshed when locale is changed
```
Comment (pierallard) [inline CHANGELOG-1.5.md]: You have to write what it change. "Fix attribute refresh when locale changed in Product Edit Form"

**Comment by pierallard** [comment]
GTM

**Comment by juliensnz** [comment]
This is a roll back of a bug fix made in 1.4. please check with @BitOne

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeOptionController.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    /**
+     * Return the attribute option array
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function getAction(Request $request)
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeOptionController.php]: should take the attribute id

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeOptionController.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        $query  = $request->query;
+        $search = $query->get('search');
+
+        $attributeOptions = $this->attributeOptionSearchableRepo->findBySearch($search, $query->get('options', []));
+
+        $normalizedAttributeOptions = [];
+        foreach ($attributeOptions as $attributeOption) {
+            $normalizedAttributeOptions[$attributeOption->getCode()] = $this->normalizer->normalize($attributeOption, 'json', ['onlyActivatedLocales' => true]);
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeOptionController.php]: do you really need to index them by code? We should not if we want to build a proper Rest API

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeOptionController.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * @param AttributeOptionSearchableRepository $attributeOptionSearchableRepo
+     * @param NormalizerInterface                 $normalizer
+     */
+    public function __construct(
+        AttributeOptionSearchableRepository $attributeOptionSearchableRepo,
+        NormalizerInterface $normalizer
+    ) {
+        $this->attributeOptionSearchableRepo = $attributeOptionSearchableRepo;
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeOptionController.php]: you don't align here and you align in the next file, please at least, be concistent

File: src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeOptionSearchableRepository.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    public function findBySearch($search = null, array $options = [])
+    {
+        $qb = $this->entityManager->createQueryBuilder();
+
+        $qb->select('o')
+            ->addSelect('v')
+            ->from($this->entityName, 'o')
+            ->leftJoin('o.optionValues', 'v')
+            ->andWhere('o.attribute=:attribute')
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeOptionSearchableRepository.php]: add spaces arround =

File: src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeOptionSearchableRepository.php [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            ->orderBy('o.sortOrder')
+            ->setParameter('attribute', $options['collectionId']);
+
+        if ($search) {
+            $qb->andWhere('v.value like :search OR o.code LIKE :search')
+                ->setParameter('search', "%$search%");
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeOptionSearchableRepository.php]: please use simple quote and concatenation :)

File: src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeOptionSearchableRepository.php [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    protected function applyQueryOptions(QueryBuilder $qb, array $options)
+    {
+        if (isset($options['identifiers']) && is_array($options['identifiers']) && !empty($options['identifiers'])) {
+            $qb->andWhere('o.code in (:codes)');
+            $qb->setParameter('codes', $options['identifiers']);
+        }
+
+        if (isset($options['locale']) && null !== $options['locale']) {
+            $qb->andWhere('v.locale=:locale');
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeOptionSearchableRepository.php]: space around =

File: src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeOptionSearchableRepository.php [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    /**
+     * @param QueryBuilder $qb
+     * @param array        $options
+     *
+     * @return QueryBuilder
+     */
+    protected function applyQueryOptions(QueryBuilder $qb, array $options)
+    {
+        if (isset($options['identifiers']) && is_array($options['identifiers']) && !empty($options['identifiers'])) {
+            $qb->andWhere('o.code in (:codes)');
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeOptionSearchableRepository.php]: does it handle the case of codes being empty?

Comment (momoss) [inline src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeOptionSearchableRepository.php]: There is a check on empty  ```!empty($options['identifiers'])```

Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeOptionSearchableRepository.php]: 👍

**Pull request merged**